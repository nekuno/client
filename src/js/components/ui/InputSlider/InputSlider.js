import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './InputSlider.scss';
import { createSliderWithTooltip } from 'rc-slider';
import Slider from 'rc-slider';
import 'rc-slider/assets/index.css';
const SliderCustom = createSliderWithTooltip(Slider);

export default class InputSlider extends Component {

    static propTypes = {
        handleChangeInputSlider : PropTypes.func.isRequired,
        data                    : PropTypes.number,
        color                   : PropTypes.string,
    };

    handleChangeInputSlider(slider) {
        this.props.handleChangeInputSlider(slider);
    }

    render() {
        const {children, data, color} = this.props;

        let rangeProps = {
            marks: {10:'10', 50:'50', 100:'100', 250:'250', 500:'500'},
            min: 10,
            max: 500,
            step: 10,
            defaultValue: data,
            handleStyle: [{background: color, color: color, borderColor: color, boxShadow: 'none'}],
            trackStyle: [{background: color, color: color, borderColor: color}],
            railStyle: {background: color, borderColor: color},
            dotStyle: {background: color, borderColor: color},
            activeDotStyle: {background: color, borderColor: color},
            onAfterChange: this.handleChangeInputSlider.bind(this)
        };

        return (
            <div className={styles.inputSlider}>
                <SliderCustom {...rangeProps}/>
            </div>
        );
    }
}