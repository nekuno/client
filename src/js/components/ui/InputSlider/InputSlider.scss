@import '../../../../scss/_partials/colors';

.input-slider {
  padding: 0 25px 25px 25px;
}