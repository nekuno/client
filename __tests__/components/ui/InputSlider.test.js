import React from 'react';
import InputSlider from '../../../src/js/components/ui/InputSlider/InputSlider.js';
import styles from '../../../src/js/components/ui/InputSlider/InputSlider.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test InputSlider component', () => {
    const mockFn = jest.fn();
    const inputslider = shallow(<InputSlider handleChangeInputSlider={mockFn}>Lorem ipsum</InputSlider>);

    it('should be defined', () => {
        expect(inputslider).toBeDefined();
    });
    it('should render correctly', () => {
        expect(inputslider).toMatchSnapshot();
    });
});
