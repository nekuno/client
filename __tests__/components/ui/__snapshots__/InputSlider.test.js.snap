// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputSlider component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputSlider
    handleChangeInputSlider={[MockFunction]}
  >
    Lorem ipsum
  </InputSlider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ComponentWrapper
        activeDotStyle={
          Object {
            "background": undefined,
            "borderColor": undefined,
          }
        }
        dotStyle={
          Object {
            "background": undefined,
            "borderColor": undefined,
          }
        }
        handleStyle={
          Array [
            Object {
              "background": undefined,
              "borderColor": undefined,
              "boxShadow": "none",
              "color": undefined,
            },
          ]
        }
        marks={
          Object {
            "10": "10",
            "100": "100",
            "250": "250",
            "50": "50",
            "500": "500",
          }
        }
        max={500}
        min={10}
        onAfterChange={[Function]}
        railStyle={
          Object {
            "background": undefined,
            "borderColor": undefined,
          }
        }
        step={10}
        tipFormatter={[Function]}
        tipProps={Object {}}
        trackStyle={
          Array [
            Object {
              "background": undefined,
              "borderColor": undefined,
              "color": undefined,
            },
          ]
        }
      />,
      "className": "inputSlider",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeDotStyle": Object {
          "background": undefined,
          "borderColor": undefined,
        },
        "defaultValue": undefined,
        "dotStyle": Object {
          "background": undefined,
          "borderColor": undefined,
        },
        "handleStyle": Array [
          Object {
            "background": undefined,
            "borderColor": undefined,
            "boxShadow": "none",
            "color": undefined,
          },
        ],
        "marks": Object {
          "10": "10",
          "100": "100",
          "250": "250",
          "50": "50",
          "500": "500",
        },
        "max": 500,
        "min": 10,
        "onAfterChange": [Function],
        "railStyle": Object {
          "background": undefined,
          "borderColor": undefined,
        },
        "step": 10,
        "tipFormatter": [Function],
        "tipProps": Object {},
        "trackStyle": Array [
          Object {
            "background": undefined,
            "borderColor": undefined,
            "color": undefined,
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ComponentWrapper
          activeDotStyle={
            Object {
              "background": undefined,
              "borderColor": undefined,
            }
          }
          dotStyle={
            Object {
              "background": undefined,
              "borderColor": undefined,
            }
          }
          handleStyle={
            Array [
              Object {
                "background": undefined,
                "borderColor": undefined,
                "boxShadow": "none",
                "color": undefined,
              },
            ]
          }
          marks={
            Object {
              "10": "10",
              "100": "100",
              "250": "250",
              "50": "50",
              "500": "500",
            }
          }
          max={500}
          min={10}
          onAfterChange={[Function]}
          railStyle={
            Object {
              "background": undefined,
              "borderColor": undefined,
            }
          }
          step={10}
          tipFormatter={[Function]}
          tipProps={Object {}}
          trackStyle={
            Array [
              Object {
                "background": undefined,
                "borderColor": undefined,
                "color": undefined,
              },
            ]
          }
        />,
        "className": "inputSlider",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeDotStyle": Object {
            "background": undefined,
            "borderColor": undefined,
          },
          "defaultValue": undefined,
          "dotStyle": Object {
            "background": undefined,
            "borderColor": undefined,
          },
          "handleStyle": Array [
            Object {
              "background": undefined,
              "borderColor": undefined,
              "boxShadow": "none",
              "color": undefined,
            },
          ],
          "marks": Object {
            "10": "10",
            "100": "100",
            "250": "250",
            "50": "50",
            "500": "500",
          },
          "max": 500,
          "min": 10,
          "onAfterChange": [Function],
          "railStyle": Object {
            "background": undefined,
            "borderColor": undefined,
          },
          "step": 10,
          "tipFormatter": [Function],
          "tipProps": Object {},
          "trackStyle": Array [
            Object {
              "background": undefined,
              "borderColor": undefined,
              "color": undefined,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
